src/neuralnet/tally.cpp

```diff
@@ -0,0 +1,1190 @@
+    {
+        return 0.0;
+    }
+
+    double PaymentPerDay(const ResearchAccount& account) const override
+    {
+        return 0.0;
+    }
+
+    double PaymentPerDayLimit(const ResearchAccount& account) const override
+    {
+        return 0.0;
+    }
+
+    bool ExceededRecentPayments(const ResearchAccount& account) const override
+    {
+        return false;
+    }
+
+    double ExpectedDaily(const ResearchAccount& account) const override
+    {
+        return 0.0;
+    }
+
+    int64_t RawAccrual(const ResearchAccount& account) const override
+    {
+        return 0.0;
+    }
+
+    int64_t Accrual(const ResearchAccount& account) const override
+    {
+        return 0.0;
+    }
+}; // NullAccrualComputer
+
+//!
+//! \brief An accrual calculator for a CPID that never earned a research reward
+//! before.
+//!
+class NewbieAccrualComputer : public NullAccrualComputer
+{
+    // See IAccrualComputer for inherited API documentation.
+
+public:
+    //!
+    //! \brief Initialze an accrual calculator for a CPID that never earned
+    //! a research reward before.
+    //!
+    //! \param cpid           CPID to calculate research accrual for.
+    //! \param payment_time   Time of payment to calculate rewards at.
+    //! \param magnitude_unit Current network magnitude unit to factor in.
+    //!
+    NewbieAccrualComputer(
+        const Cpid cpid,
+        const int64_t payment_time,
+        const double magnitude_unit)
+        : m_cpid(cpid)
+        , m_payment_time(payment_time)
+        , m_magnitude_unit(magnitude_unit)
+    {
+    }
+
+    double MagnitudeUnit() const override
+    {
+        return m_magnitude_unit;
+    }
+
+    int64_t AccrualAge(const ResearchAccount& account) const override
+    {
+        const int64_t beacon_time = BeaconTimeStamp(m_cpid.ToString());
+
+        if (beacon_time == 0) {
+            return 0;
+        }
+
+        return m_payment_time - beacon_time;
+    }
+
+    double AccrualDays(const ResearchAccount& account) const override
+    {
+        return AccrualAge(account) / ONE_DAY_IN_SECONDS;
+    }
+
+    double AverageMagnitude(const ResearchAccount& account) const override
+    {
+        return account.m_magnitude;
+    }
+
+    double PaymentPerDayLimit(const ResearchAccount& account) const override
+    {
+        return 500.00;
+    }
+
+    bool ExceededRecentPayments(const ResearchAccount& account) const override
+    {
+        return RawAccrual(account) > PaymentPerDayLimit(account) * COIN;
+    }
+
+    double ExpectedDaily(const ResearchAccount& account) const override
+    {
+        return account.m_magnitude * m_magnitude_unit;
+    }
+
+    int64_t RawAccrual(const ResearchAccount& account) const override
+    {
+        return AccrualDays(account) * account.m_magnitude * m_magnitude_unit * COIN;
+    }
+
+    int64_t Accrual(const ResearchAccount& account) const override
+    {
+        constexpr int64_t six_months = ONE_DAY_IN_SECONDS * 30 * 6;
```

Comment:
I thought about doing that... since each implementation of `IAccrualCalculator` could, in theory, rely on its own definition of a cutoff--either in time or in height--I didn't want to hard-code a general constant for these cases.

Reply:
Ah I get it. You would do another implementation of the interface rather than generalize this specifically. Makes sense.